articles/bottom-navigation-bar-in-android/index.md

```diff
@@ -0,0 +1,290 @@
+```
+A NavHost is an empty container used to display destinations from the navigation graph
+
+**Note : An error for `app:menu="@menu/bottom_nav_menu"` is seen.This is because it does not exist. To solve this, click `Alt`+ `Enter` and select "Create resource file for bottom_nav_menu.xml"**
+
+
+![Create a resource file](/engineering-education/bottom-navigation-bar-in-android/createmenu.jpg)
+
+This menu will pop up.
+
+![Menu](/engineering-education/bottom-navigation-bar-in-android/menu.jpg)
+
+Select *OK* and the error is fixed.
+
+### Step 4 -Creating A Fragment.
+A **Fragment** is a sub-activity. Fragments are used to simplify the reuse of components and logic in different layouts. First, navigate to the java directory and right-click. Select *New*. Click *Fragment* and select *Fragment(Blank)*. Name the Fragment and select *Finish*.
+
+
+### Step 5 - Adding Details and Icons to the Bottom Navigation View
+First, let us add the icons required. In the `res/drawable` directory, right-click the `drawable` folder. Select *new* and then select **Vector Asset**.
+
+![Vector Asset](/engineering-education/bottom-navigation-bar-in-android/vector.jpg)
+
+* **_Asset type_** - provides an option of using a clipart or uploading a local file.
+* **_Name_** - name of the icon.
+* **_Size_**- the dimensions of the clipart
+* **_Color_** - provides color options for the clipart.
+* **_Opacity_** - deals with the visibility of the icon.
+
+Select on the icon next to clipart. A list of different cliparts is displayed.
+
+![Clipart List](/engineering-education/bottom-navigation-bar-in-android/list.jpg)
+
+One can search for a clipart in the search bar and choose whether the design of the clipart should be filled or outlined. We will use the outlined design.
+
+![Favorites](/engineering-education/bottom-navigation-bar-in-android/favorites.jpg)
+
+Once a clipart is selected, click *OK* then *Next* and finally *Finish*. Open the menu file in `res/menu/ bottom_navigation_menu` and add the following lines of code:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <item
+        android:id="@+id/navigation_home"
+        android:icon="@drawable/ic_outline_home_24"
+        android:title="@string/title_home" />
+    <item
+        android:id="@+id/navigation_search"
+        android:icon="@drawable/ic_outline_search_24"
+        android:title="@string/title_search" />
+    <item
+        android:id="@+id/navigation_favourites"
+        android:icon="@drawable/ic_baseline_favorite_border_24"
+        android:title="@string/title_favourites" />
+    <item
+        android:id="@+id/navigation_profile"
+        android:icon="@drawable/ic_baseline_perm_identity_24"
+        android:title="@string/title_profile" />
+
+</menu>
+
+```
+The **menu** file serves as a container for menu items. **Item** represents one item on the menu. Each item has a title and an icon.
+
+#### Step 6 - Let's Code.
```

Comment:
Didn't we start coding before this step? You can sth else since XML is also part of the code....